Sources/NIOSSH/SSHPacketSerializer.swift

```diff
@@ -47,40 +47,61 @@ struct SSHPacketSerializer {
                 preconditionFailure("only .version message is allowed at this point")
             }
         case .cleartext:
-            let index = buffer.writerIndex
+            buffer.writeSSHPacket(message: message, lengthEncrypted: true, blockSize: 8)
+            self.sequenceNumber &+= 1
+        case .encrypted(let protection):
+            let index = buffer.readerIndex
+            buffer.moveReaderIndex(to: buffer.writerIndex)
+            buffer.writeSSHPacket(message: message, lengthEncrypted: protection.lengthEncrypted, blockSize: protection.cipherBlockSize)
```

Comment:
another downside maybe that transport encryption now used in two calls, for packet and for encrytion